pkg/webhook/certificate_builder.go

```diff
@@ -0,0 +1,112 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	genCertFormat = "%s --service %s --namespace %s --certDir %s"
+)
+
+type CertificateBuilder struct {
+	log logr.Logger
+	client.Client
+}
+
+func NewCertificateBuilder(c client.Client, log logr.Logger) *CertificateBuilder {
+	ch := &CertificateBuilder{Client: c, log: log}
+	return ch
+}
+
+func (c *CertificateBuilder) BuildAndSyncCABundle(svcName, webhookName, cerPath string) error {
+
+	ns, err := utils.GetEnvByKey(common.MyPodNamespace)
+	if err != nil {
+		return errors.Wrapf(err, "get namespace from env failed, env key:%s", common.MyPodNamespace)
+	}
+	c.log.Info("start generate certificate", "service", svcName, "namespace", ns, "cert dir", cerPath)
+
+	ca, err := c.genCA(ns, svcName, common.CertificationGenerateFile, cerPath)
+	if err != nil {
+		return err
+	}
+
+	return c.PatchCABundle(webhookName, ca)
+}
+
+func (c *CertificateBuilder) genCA(ns, svc, certFile, certPath string) ([]byte, error) {
```

Comment:
please add code annotation for these three functions

Reply:
done